visualization/src/point_picking_event.cpp

```diff
@@ -167,49 +189,67 @@ pcl::visualization::PointPickingCallback::performSinglePick (
     {
-      vtkSmartPointer<vtkIdTypeArray> global_ids = vtkIdTypeArray::New ();
-      global_ids->SetNumberOfValues (picker->GetDataSet ()->GetNumberOfPoints ());
-      for (vtkIdType i = 0; i < picker->GetDataSet ()->GetNumberOfPoints (); ++i)
-        global_ids->SetValue (i,i);
+      vtkActor* actor = vtkActor::SafeDownCast(prop);
+      if(actor)
+      {
+        actors_->AddItem(actor);
+
+        std::vector<int> actorIndices;
+        vtkPolyData* pd = vtkPolyData::SafeDownCast(actor->GetMapper()->GetInput());
+
+        if(pd->GetPointData()->HasArray("Indices"))
+            pd->GetPointData()->RemoveArray("Indices");
 
-      points->GetPointData ()->SetGlobalIds (global_ids);
-    }
 
-    vtkPlanes* frustum = picker->GetFrustum ();
+        vtkSmartPointer<vtkIdTypeArray> IDs = vtkSmartPointer<vtkIdTypeArray>::New();
+        IDs->SetNumberOfComponents(1);
+        IDs->SetName("Indices");
+        for(vtkIdType i = 0; i < pd->GetNumberOfPoints(); i++)
+          IDs->InsertNextValue(i);
 
-    vtkSmartPointer<vtkExtractGeometry> extract_geometry = vtkSmartPointer<vtkExtractGeometry>::New ();
-    extract_geometry->SetImplicitFunction (frustum);
+        pd->GetPointData()->AddArray(IDs);
 
-    extract_geometry->SetInputData (picker->GetDataSet ());
+        vtkPlanes* frustum = picker->GetFrustum();
 
-    extract_geometry->Update ();
+        vtkSmartPointer<vtkExtractGeometry> extract_geometry = vtkSmartPointer<vtkExtractGeometry>::New();
+        extract_geometry->SetImplicitFunction(frustum);
+  #if VTK_MAJOR_VERSION < 6
+        extract_geometry->SetInput(pd);
+  #else
+        extract_geometry->SetInputData(pd);
+  #endif
 
-    vtkSmartPointer<vtkVertexGlyphFilter> glyph_filter = vtkSmartPointer<vtkVertexGlyphFilter>::New ();
-    glyph_filter->SetInputConnection (extract_geometry->GetOutputPort ());
-    glyph_filter->Update ();
+        extract_geometry->Update();
 
-    vtkPolyData* selected = glyph_filter->GetOutput ();
-    vtkIdTypeArray* ids = vtkIdTypeArray::SafeDownCast(selected->GetPointData ()->GetGlobalIds ());
-    assert (ids);
-    indices.reserve (ids->GetNumberOfTuples ());
+        vtkSmartPointer<vtkVertexGlyphFilter> glyph_filter = vtkSmartPointer<vtkVertexGlyphFilter>::New ();
+        glyph_filter->SetInputConnection (extract_geometry->GetOutputPort ());
+        glyph_filter->Update ();
 
-    for(vtkIdType i = 0; i < ids->GetNumberOfTuples (); i++)
-      indices.push_back (static_cast<int> (ids->GetValue (i)));
+        vtkPolyData* selected = glyph_filter->GetOutput ();
 
-    return (static_cast<int> (selected->GetNumberOfPoints ()));
+        vtkIdTypeArray* GlobalIDs  = vtkIdTypeArray::SafeDownCast(selected->GetPointData()->GetArray("Indices"));
+        for (vtkIdType i = 0; i < selected->GetNumberOfPoints (); i++)
+        {
+          actorIndices.push_back(static_cast<int>(GlobalIDs->GetValue(i)));
+          pt_numb++;
```

Comment:
`pt_numb += actorIndices.size()` seems equivalent

Reply:
pt_numb should be number of point from all actors, actorIndices are only for one actor

Reply:
Yes, but each actor causes one item to be added to actor indices, so doing `pt_num++` N times and doing `pt_numb += N` is functionally equivalent.